### ISSUE TITLE
JS -- String.prototype.repeat() spec. inconsistency

### ISSUE THREAD

**Issue by titouandk** [Codewars/codewars.com#548]
**Description**

On http://www.codewars.com/, the `String.prototype.repeat()` function do not follow ES6 spec.
The bug occures when `-1` is passed to the method.

**Steps to reproduce**
- Open a browser
- Go on http://www.codewars.com/ (on others sites, `String.prototype.repeat()` works fine)
- In the browser's JS console, typing `"a".repeat(-1)` returns `""`. Negative numbers should raise a `RangeError` instead.

SPEC http://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.repeat

**Comment by jhoffner** [comment]
We can look into this the next time BabelJS/Node is updated on the runner image.

**Comment by kazk** [comment]
Node v6 confirms to the spec and throws `RangeError: Invalid count value`.
Node v0.10 with or without Babel doesn't have `.repeat`.

---

> - Go on http://www.codewars.com/ (**on others sites**, String.prototype.repeat() works fine)
> - In the **JS console**, "a".repeat(-1) returns "". Negative numbers should raise a RangeError instead.

Is this "JS console" browser's developer console? If so, I just tried and confirmed.

```js
"a".repeat(-1); //=> ""
String.prototype.repeat + ''; //=> "function (e){return Array(e+1).join(this)}"
```

Viewing the minified `/assets/application.js`, I found many modifications to native types.

---

@jhoffner changing builtin methods unconditionally like this will cause many issues and they're really hard to debug. These modifications might also cause issues using libraries.
I'd suggest not to do this (`String.prototype.repeat` is widely supported now and others can be written as simple functions). If not, at least test for the existence before overwriting:

```js
if (typeof String.prototype.repeat != 'function') String.prototype.repeat = function(n) {};
```

This shouldn't break anything unless the code relies on behaviors against the spec.
